cbmc/proofs/IotMqtt_UnsubscribeAsync/Makefile

```diff
@@ -0,0 +1,117 @@
+ABSTRACTIONS += --remove-function-body _validateQos
+ABSTRACTIONS += --remove-function-body IotNetworkInterfaceClose
+ABSTRACTIONS += --remove-function-body IotNetworkInterfaceReceive
+#ABSTRACTIONS += --remove-function-body _IotMqtt_PublishSetDup
+#ABSTRACTIONS += --remove-function-body _IotMqtt_RemoveSubscriptionByPacket
+#ABSTRACTIONS += --remove-function-body _checkRetryLimit
+#ABSTRACTIONS += --remove-function-body _scheduleNextRetry
+
+################################################################
+
+OBJS += $(ENTRY)_harness.goto
+OBJS += $(MQTT)/cbmc/proofs/mqtt_state.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_api.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_helper.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_operation.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_serialize.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_subscription.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_validate.goto
+
+DEF += -DCBMC_STUB_REMOVEALLMATCHES=1
+DEF += -DCBMC_UNSUBSCRIBEASYNC_PROOF=1
+
+# allow mqtt mallocs to fail for coverage
+DEF += -include mqtt_state.h
+DEF += -DIotMqtt_MallocOperation=malloc_can_fail
+DEF += -DIotMqtt_MallocMessage=malloc_can_fail
+
+################################################################
+# Loop unwinding
+
+LOOP += harness.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _IotMqtt_RemoveSubscriptionByTopicFilter.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _IotMqtt_SerializeUnsubscribe.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _IotMqtt_SerializeUnsubscribeCommon.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _encodeRemainingLength.0:2
+LOOP += _IotMqtt_SubscriptionPacketSize.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _validateSubscription.0:$(TOPIC_LENGTH_MAX)
+LOOP += allocate_IotMqttSubscriptionArray.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += valid_IotMqttSubscriptionArray.0:$(SUBSCRIPTION_COUNT_MAX)
+
+LOOP += valid_IotMqttSubscriptionList.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += valid_IotMqttOperationList.0:$(OPERATION_COUNT_MAX)
+LOOP += _IotMqtt_ValidateSubscriptionList.0:$(SUBSCRIPTION_COUNT_MAX)
+
+LOOP += strncmp.0:$(TOPIC_LENGTH_MAX)
+
+LOOP += IotListDouble_FindFirstMatch.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += IotListDouble_FindFirstMatch\$$link1.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += IotListDouble_FindFirstMatch\$$link2.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += IotListDouble_FindFirstMatch\$$link5.0:$(SUBSCRIPTION_COUNT_MAX)
+
+LOOP += IotListDouble_RemoveAllMatches.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_RemoveAllMatches\$$link1.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_RemoveAllMatches\$$link2.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_RemoveAllMatches\$$link3.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_RemoveAllMatches\$$link4.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_RemoveAllMatches\$$link5.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_RemoveAllMatches\$$link6.0:$(SUBSCRIPTION_LIST_MAX)
```

Comment:
Woow, do we need all of them?

Reply:
Don't know, but they have all appeared at various points in proof development, leaving them make the proof less brittle to future changes.